app/models/reviewable_history.rb

```diff
@@ -4,16 +4,22 @@ class ReviewableHistory < ActiveRecord::Base
   belongs_to :reviewable
   belongs_to :created_by, class_name: 'User'
 
-  def self.types
-    @types ||= Enum.new(
-      created: 0,
-      transitioned: 1,
-      edited: 2,
-      claimed: 3,
-      unclaimed: 4
-    )
+  enum status: %i[pending approved rejected ignored deleted]
+  enum reviewable_history_type: %i[created transitioned edited claimed unclaimed]
+
+  after_commit :compute_user_stats
+
+  # Backward compatibility
+  class << self
+    alias types reviewable_history_types
   end
 
+  private
+
+  def compute_user_stats
+    return unless (created? && pending?) || (transitioned? && !pending?)
```

Comment:
We try to avoid using `unless` with a "complicated" condition (aka `&&` or `||`) as it's not always easy to understand it.

You can either change the condition to use `if` instead or you could split this into 2 lines.